content/shmem_ctx.tex

```diff
@@ -0,0 +1,133 @@
+\color{ForestGreen}
+
+All \openshmem RMA, AMO, and memory ordering routines are
+performed on a communication context.  The communication context defines an
+independent ordering and completion environment, allowing users to manage the
+overlap of communication with computation and also to manage communication
+operations performed by separate threads within a multithreaded \ac{PE}.  For
+example, in single-threaded environments, contexts can be used to pipeline
+communication and computation.  In multithreaded environments, contexts can
+additionally provide thread isolation, eliminating overheads resulting from
+thread interference within the communication subsystem.
+
+Context handles are of type \CTYPE{shmem\_ctx\_t} and are valid for
+language-level assignment and comparison.  A handle to the desired context is
+passed as an argument in the \Cstd \CTYPE{shmem\_ctx\_*} and type-generic API
+routines.  API routines that do not accept a context argument operate on the
+default context.  The default context can be used explicitly through the
+\const{SHMEM\_CTX\_DEFAULT} handle.
+
+\subsubsection{\textbf{SHMEM\_CTX\_CREATE}}
+\label{subsec:shmem_ctx_create}
+\apisummary{
+    Create a communication context.
+}
+
+\begin{apidefinition}
+
+\begin{Csynopsis}
+int shmem_ctx_create(long options, shmem_ctx_t *ctx);
+\end{Csynopsis}
```

Comment:
Typically, when I see a creation API of this style, the created object is the first argument, not the last. Is there a particular reason this prototype was chosen?